// fake-store/dist/cjs/fake-store.cjs.entry.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-9ee2f551.js');

const fakeStoreCss = "#card{width:20rem;height:fit-content;border:0.05rem solid rgb(var(--primary-rgb, var(--ion-color-primary-rgb, (156, 9, 156))));border-radius:0.5rem;margin:0.8rem;box-shadow:2px 2px 6px 2px rgba(var(--primary-rgb, var(--ion-color-primary-rgb, (156, 9, 156))), 0.699);background:var(--card-backgroun, var(--ion-card-backgroun, white));padding-bottom:0}#card img{width:12rem;height:13rem;margin-left:auto;margin-right:auto;margin-top:1.5rem;display:block;transform:scale(0.9)}#card #title{font-family:sans-serif;text-align:center;color:rgb(var(--primary-rgb, var(--ion-color-primary-rgb, (156, 9, 156))));padding:0 1.25rem;height:2.3rem;overflow:hidden;text-overflow:ellipsis;display:-webkit-box;-webkit-line-clamp:2;-webkit-box-orient:vertical}#card #txt{font-family:sans-serif;text-align:center;color:rgb(var(--primary-rgb, var(--ion-color-primary-rgb, (156, 9, 156))))}#card .cart-btn{width:20rem;height:3rem;border:1px white;border-radius:0 0 0.5rem 0.5rem}#card .cart-btn:active,#card .clear-btn:active{background:rgba(var(--secondary-rgb, var(--ion-color-secondary-rgb, (204, 201, 201))), 0.808);transition:all 0.1s ease-out}.icon-cart{width:2rem;height:2rem;margin:0.2rem 0;fill:rgb(var(--primary-rgb, var(--ion-color-primary-rgb, (156, 9, 156))));color:rgb(var(--primary-rgb, var(--ion-color-primary-rgb, (156, 9, 156))))}.clear-btn{position:absolute;width:2.5rem;height:2.5rem;border:1px transparent;border-radius:1.5rem;transform:translateX(17.3rem) translateY(0.2rem)}.clear-icon{width:2rem;height:2rem;margin:0.25rem;fill:rgb(var(--primary-rgb, var(--ion-color-primary-rgb, (156, 9, 156))));color:rgb(var(--primary-rgb, var(--ion-color-primary-rgb, (156, 9, 156))))}#rate-icons{position:absolute;padding-top:0.25rem;padding-left:0.25rem;z-index:10}#rate-icons p{padding:0;margin:0}#start{width:1.25rem;height:1.25rem;margin:0.2rem;fill:rgb(var(--primary-rgb, var(--ion-color-primary-rgb, (156, 9, 156))));color:rgb(var(--primary-rgb, var(--ion-color-primary-rgb, (156, 9, 156))))}";

let FakeStore = class {
  constructor(hostRef) {
    index.registerInstance(this, hostRef);
    this.addedTocart = index.createEvent(this, "addedToCart", 7);
    this.removedTocart = index.createEvent(this, "removedToCart", 7);
    this.addedToFavs = index.createEvent(this, "addedToFavs", 7);
    this.removedToFavs = index.createEvent(this, "removedToFavs", 7);
  }
  _getImg() {
    return this.prodImage;
  }
  _getTitle() {
    return this.prodTitle;
  }
  _getPrice() {
    return this.prodPrice.toFixed(2);
  }
  _getRate() {
    return this.prodRate;
  }
  _addToCart() {
    this.inCart = !this.inCart;
    if (this.inCart) {
      this.addedTocart.emit(this.prodId);
    }
    else {
      this.removedTocart.emit(this.prodId);
    }
  }
  _addToFavs() {
    this.inFavorites = !this.inFavorites;
    if (this.inFavorites) {
      this.addedToFavs.emit(this.prodId);
    }
    else {
      this.removedToFavs.emit(this.prodId);
    }
  }
  render() {
    let cartIcon = null;
    if (this.inCart) {
      cartIcon = (index.h("svg", { xmlns: "http://www.w3.org/2000/svg", class: "icon-cart", viewBox: "0 0 512 512" }, index.h("title", null, "Cart"), index.h("circle", { cx: "176", cy: "416", r: "32" }), index.h("circle", { cx: "400", cy: "416", r: "32" }), index.h("path", { d: "M456.8 120.78a23.92 23.92 0 00-18.56-8.78H133.89l-6.13-34.78A16 16 0 00112 64H48a16 16 0 000 32h50.58l45.66 258.78A16 16 0 00160 368h256a16 16 0 000-32H173.42l-5.64-32h241.66A24.07 24.07 0 00433 284.71l28.8-144a24 24 0 00-5-19.93z" })));
    }
    else {
      cartIcon = (index.h("svg", { xmlns: "http://www.w3.org/2000/svg", class: "icon-cart", viewBox: "0 0 512 512" }, index.h("title", null, "Cart"), index.h("circle", { cx: "176", cy: "416", r: "16", fill: "none", stroke: "currentColor", "stroke-linecap": "round", "stroke-linejoin": "round", "stroke-width": "32" }), index.h("circle", { cx: "400", cy: "416", r: "16", fill: "none", stroke: "currentColor", "stroke-linecap": "round", "stroke-linejoin": "round", "stroke-width": "32" }), index.h("path", { fill: "none", stroke: "currentColor", "stroke-linecap": "round", "stroke-linejoin": "round", "stroke-width": "32", d: "M48 80h64l48 272h256" }), index.h("path", { d: "M160 288h249.44a8 8 0 007.85-6.43l28.8-144a8 8 0 00-7.85-9.57H128", fill: "none", stroke: "currentColor", "stroke-linecap": "round", "stroke-linejoin": "round", "stroke-width": "32" })));
    }
    let heartIcon = null;
    if (this.inFavorites) {
      heartIcon = (index.h("svg", { xmlns: "http://www.w3.org/2000/svg", class: "clear-icon", viewBox: "0 0 512 512" }, index.h("title", null, "Heart"), index.h("path", { d: "M256 448a32 32 0 01-18-5.57c-78.59-53.35-112.62-89.93-131.39-112.8-40-48.75-59.15-98.8-58.61-153C48.63 114.52 98.46 64 159.08 64c44.08 0 74.61 24.83 92.39 45.51a6 6 0 009.06 0C278.31 88.81 308.84 64 352.92 64c60.62 0 110.45 50.52 111.08 112.64.54 54.21-18.63 104.26-58.61 153-18.77 22.87-52.8 59.45-131.39 112.8a32 32 0 01-18 5.56z" })));
    }
    else {
      heartIcon = (index.h("svg", { xmlns: "http://www.w3.org/2000/svg", class: "clear-icon", viewBox: "0 0 512 512" }, index.h("title", null, "Heart"), index.h("path", { d: "M352.92 80C288 80 256 144 256 144s-32-64-96.92-64c-52.76 0-94.54 44.14-95.08 96.81-1.1 109.33 86.73 187.08 183 252.42a16 16 0 0018 0c96.26-65.34 184.09-143.09 183-252.42-.54-52.67-42.32-96.81-95.08-96.81z", fill: "none", stroke: "currentColor", "stroke-linecap": "round", "stroke-linejoin": "round", "stroke-width": "32" })));
    }
    let rateIcons = [];
    for (let i = 0; i < 5; i++) {
      if (this.prodRate >= 0.8 + i) {
        rateIcons.push(index.h("svg", { xmlns: "http://www.w3.org/2000/svg", id: "start", viewBox: "0 0 512 512" }, index.h("title", null, "Star"), index.h("path", { d: "M394 480a16 16 0 01-9.39-3L256 383.76 127.39 477a16 16 0 01-24.55-18.08L153 310.35 23 221.2a16 16 0 019-29.2h160.38l48.4-148.95a16 16 0 0130.44 0l48.4 149H480a16 16 0 019.05 29.2L359 310.35l50.13 148.53A16 16 0 01394 480z" })));
      }
      else if (this.prodRate >= 0.3 + i) {
        rateIcons.push(index.h("svg", { xmlns: "http://www.w3.org/2000/svg", id: "start", viewBox: "0 0 512 512" }, index.h("title", null, "Star Half"), index.h("path", { d: "M480 208H308L256 48l-52 160H32l140 96-54 160 138-100 138 100-54-160z", fill: "none", stroke: "currentColor", "stroke-linejoin": "round", "stroke-width": "32" }), index.h("path", { d: "M256 48v316L118 464l54-160-140-96h172l52-160z" })));
      }
      else {
        rateIcons.push(index.h("svg", { xmlns: "http://www.w3.org/2000/svg", id: "start", viewBox: "0 0 512 512" }, index.h("title", null, "Star"), index.h("path", { d: "M480 208H308L256 48l-52 160H32l140 96-54 160 138-100 138 100-54-160z", fill: "none", stroke: "currentColor", "stroke-linejoin": "round", "stroke-width": "32" })));
      }
    }
    return (index.h("div", { id: "card" }, index.h("buttn", { class: "clear-btn", onClick: this._addToFavs.bind(this) }, heartIcon), index.h("div", { id: "rate-icons" }, rateIcons, index.h("p", { id: "txt" }, index.h("small", null, this._getRate()))), index.h("img", { src: this._getImg(), alt: "Foto" }), index.h("p", { id: "title" }, index.h("strong", null, this._getTitle())), index.h("p", { id: "txt" }, " $", this._getPrice(), " MXN"), index.h("button", { class: "cart-btn", onClick: this._addToCart.bind(this) }, cartIcon)));
  }
};
FakeStore.style = fakeStoreCss;

exports.fake_store = FakeStore;
